Dialog/balloon.gd

```diff
@@ -0,0 +1,176 @@
+		responses_menu.next_action = next_action
+
+
+func _unhandled_input(_event: InputEvent) -> void:
+	# Only the balloon is allowed to handle input while it's showing
+	get_viewport().set_input_as_handled()
+
+
+## Start some dialogue
+func start(dialogue_resource: DialogueResource, title: String, extra_game_states: Array = []) -> void:
+	temporary_game_states =  [self] + extra_game_states
+	is_waiting_for_input = false
+	resource = dialogue_resource
+	self.dialogue_line = await resource.get_next_dialogue_line(title, temporary_game_states)
+
+
+## Go to the next line
+func next(next_id: String) -> void:
+	self.dialogue_line = await resource.get_next_dialogue_line(next_id, temporary_game_states)
+
+
+### Signals
+
+
+func _on_mutated(_mutation: Dictionary) -> void:
+	is_waiting_for_input = false
+	will_hide_balloon = true
+	get_tree().create_timer(0.1).timeout.connect(func() -> void:
+		if will_hide_balloon:
+			will_hide_balloon = false
+			balloon.hide()
+	)
+
+
+func _on_balloon_gui_input(event: InputEvent) -> void:
+	# See if we need to skip typing of the dialogue
+	if dialogue_label.is_typing:
+		var mouse_was_clicked: bool = event is InputEventMouseButton and event.button_index == MOUSE_BUTTON_LEFT and event.is_pressed()
+		var skip_button_was_pressed: bool = event.is_action_pressed(skip_action)
+		if mouse_was_clicked or skip_button_was_pressed:
+			get_viewport().set_input_as_handled()
+			dialogue_label.skip_typing()
+			return
+
+	if not is_waiting_for_input: return
+	if dialogue_line.responses.size() > 0: return
+
+	# When there are no response options the balloon itself is the clickable thing
+	get_viewport().set_input_as_handled()
+
+	if event is InputEventMouseButton and event.is_pressed() and event.button_index == MOUSE_BUTTON_LEFT:
+		next(dialogue_line.next_id)
+	elif event.is_action_pressed(next_action) and get_viewport().gui_get_focus_owner() == balloon:
+		next(dialogue_line.next_id)
+
+# Custom function that is interacted via the .dialogue file to add a portrait to the scene
+func add_portrait(art_name: String) -> void:
+	portrait.texture = load("res://Art/%s.png" % art_name)
+
+# Custom function that is interacted via the .dialogue file to add some flavor image based on the index passed in
+func add_flavor_image(image_name: String, flavor_number: int) -> void:
+	var flavor_image: Texture2D = load("res://Art/%s.png" % image_name)
+	if (flavor_number == 1):
+		flavor_image_1.texture = flavor_image
+	elif (flavor_number == 2):
+		flavor_image_2.texture = flavor_image
+	elif (flavor_number == 3):
+		flavor_image_3.texture = flavor_image
+
+# Custom function that is interacted via the .dialogue file to add some flavor text based on the index passed in
+func add_flavor_text(flavor_text: String, flavor_number: int) -> void:
+	if (flavor_number == 1):
+		flavor_text_1.add_text(flavor_text)
+	elif (flavor_number == 2):
+		flavor_text_2.add_text(flavor_text)
+	elif (flavor_number == 3):
+		flavor_text_3.add_text(flavor_text)
```

Comment:
I think it would be best to make this a match statement instead, as it will set the example for other files

Does it also mean we need one such .gd file for each new dialogue ?

Reply:
I'll update to match.

And no, balloon.gd is used in balloon.tscn which is the packed scene that is used to bring up the dialog screen. 
[This change](https://github.com/Saplings-Projects/1M_sub/pull/92/files#diff-b7e8404905efe7afd6c8148ccc42e6f2cc45a58bc44ed0b5007a74bc3398c5daR199) will pull the packed scene and gets the dialog file we specify. In this case we get the test.dialogue. So we don't need multiple .gd scripts, we need multiple dialogue files.

Reply:
okay, but it seems to me that since we only have match against number 1 to 3, we can only have 3 flavor text. What does each flavor text represent ?
Also it would be best to rename this from balloon to something that shows it's the file managing the dialogues

Reply:
![image](https://github.com/Saplings-Projects/1M_sub/assets/5923102/fcd8e804-e056-4f40-9374-be9e844f6b92)

Each flavor text corresponds to the fixed location on the balloon. I marked it on the screenshot. This is what I was alluding to in terms of the fake "randomness" of the doodles and writings that was discussed with design. Within the .dialogue file we can call `add_flavor_image()` or `add_flavor_text()` and add whatever image/text to supplement the event we're trying to display. This also isn't limited to three, those were just three places I set those Controls on, we can very well add more if we want and update this balloon.gd file.

Reply:
oh i see, it makes more sense now